presto-main/src/test/java/com/facebook/presto/sql/planner/planPrinter/TestPlanPrinter.java

```diff
@@ -186,4 +189,29 @@ public void testDomainTextFormatting()
                 Domain.onlyNull(VARCHAR),
                 "[NULL]");
     }
+
+    @Test
+    public void testPrintNativeExecutionNode()
+    {
+        ValuesNode values = PLAN_BUILDER.values();
+        LimitNode limit = PLAN_BUILDER.limit(10, values);
+        NativeExecutionNode nativeExecution = PLAN_BUILDER.NativeExecution(limit);
+
+        PlanFragment testFragment = new PlanFragment(
+                new PlanFragmentId(0),
+                nativeExecution,
+                ImmutableSet.of(),
+                SOURCE_DISTRIBUTION,
+                ImmutableList.of(nativeExecution.getId()),
```

Comment:
Question: Why do we put this node in the table scan node list? Can we leave it empty? Or this is by design?

Reply:
It's by design, when native execution turned on, instead of tableScans, the NativeExecutionNode will be served as the only source operator during presto driver scheduling time (which this field is being used for).

Reply:
@miaoever Thanks for explaining.